iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c

```diff
@@ -1171,6 +1171,7 @@ static void setup_initialize_reconnection_mocks()
         .CopyOutArgumentBuffer_retry_action(&retry_action, sizeof(retry_action));
     STRICT_EXPECTED_CALL(tickcounter_get_current_ms(IGNORED_PTR_ARG, IGNORED_PTR_ARG));
     STRICT_EXPECTED_CALL(xio_retrieveoptions(IGNORED_PTR_ARG));
+    STRICT_EXPECTED_CALL(mqtt_client_clear_xio(IGNORED_PTR_ARG));
```

Comment:
General comment I'm putting just on this line...

This issue of was hit in field with #1666.  We should add a new test that repros that this as the customer ran into it so that we know if we fall into that case in future (and especially as code around us changes) we'll still be protected.

I get these are a fair amount of work.  I can help with advise on best way to model this if needed.

Reply:
ahhhh crap. yea that would be ideal. I'll link base with you to figure out how we can go about doing that.

Reply:
added. as discussed, this new unit test UNIT tests the disconnect scenario. it does not e2e test the scenario from the gh issue.